liam2/matching.py

```diff
@@ -8,8 +8,10 @@
 from liam2.exprbases import FilteredExpression
 from liam2.context import context_length, context_delete, context_subset, context_keep
 from liam2.utils import loop_wh_progress
-# FIXME: should be optional
-from liam2.cpartition import group_indices_nd
+try:
+    from cpartition import group_indices_nd
+except:
+    pass
```

Comment:
This will fail later if one uses algo="byvalue". You should probably define here a fake group_indices_nd which throws a nice exception (aglo="byvalue" is not available when C extensions blah blah...) when called.